### ISSUE TITLE
Question: objprint generator sup

### ISSUE THREAD

**Issue by Euraxluo** [gaogaotiantian/objprint#56]
![image](https://user-images.githubusercontent.com/34028978/158317683-6ac6132e-c68d-4f18-891e-9dc3c859969c.png)

now objprint result:
![image](https://user-images.githubusercontent.com/34028978/158317789-b212a7f7-0687-499d-9c65-adb714d216a6.png)

I've been working on state coding for closures and iterators
I think objprint might help me

**Comment by gaogaotiantian** [comment]
As of now, `objprint` is not able to show any built-in data objects that do not have `__dict__`, because it searches key/value pairs on `__dict__`. This could be changed by using `__dir__` instead, but I need to look into the consequences. The `objprint` result you saw was just the default string representation of the generator.

**Comment by mochazi** [comment]
```python
#Python3.7
#encoding=utf-8

'''objprint==0.2.1'''
from objprint import op

'''Closure Demo'''
class ClosureBase():

    def __init__(self,func):
        
        self._outer = func
        self._inner = func()
        self._result = func()()
 
@ClosureBase
def outer():
    
    def inner():
        return [1,2,3]

    return inner 

op(outer)

'''Generator Demo'''
class GeneratorBase():

    def __init__(self,func):
        
        self._generator = func(2)
        self._next = next(self._generator)
        self._next_next = next(self._generator)

@GeneratorBase
def create_counter(n):

    while True:
        yield n

        n +=1
 
op(create_counter)
```
**now objprint result:**

![code](https://user-images.githubusercontent.com/54659396/159629951-0291ccd9-4697-4f2e-ba77-28c0ce2c58dc.png)

@gaogaotiantian @Euraxluo 
**I think of a temporary solution.**

**Comment by gaogaotiantian** [comment]
I believe he needs the internal states of generators, which is not part of ``__dict__`` of the generator. I need some time to check the ``dir`` solution and I think that would be the way to go.

**Comment by gaogaotiantian** [comment]
Solved in #65
